### ISSUE TITLE
Inline images are rendering in ADO's Description field for one work item type but not another

### ISSUE THREAD

**Issue by cacti77** [solidify/jira-azuredevops-migrator#1063]
**Describe the problem**
In the source Jira server project I have Bugs and Improvements. Some of these have inline images/screenshots in their _Description_ field. When I import these issues into the target ADO project, the inline images render in the standard _Description_ field of the Improvements but not the Bugs; e.g.:
![image](https://github.com/user-attachments/assets/62d3e872-6c7a-4091-93a8-2bdf9d3a21e1)

**To Reproduce**
Extracts from the config json file:
```
  "process-template": "Basic",

  "type-map": {
    "type": [
      {
        "source": "Improvement",
        "target": "Myprocess.Improvement"
      },
      {
        "source": "Bug",
        "target": "Myprocess.Bug"
      }
    ]
  },
  "field-map": {
    "field": [
      {
        "source": "description",
        "target": "System.Description",
        "mapper": "MapRendered"
      },
```
where Myprocess is the name of my ADO process which inherits from Basic. Myprocess has Epic, Issue and Task disabled, but new work item types called Bug and Improvement. Both of these have the default _System.Description_ field but are otherwise virtually identical. Neither WIT has a _Repro Steps_ field (and so I don't refer to that in the config json file either).

When I look in the issue json files generated by jira-export you can see entries like this inside a Myprocess.Improvement's _System.Description_ field:
```
<p>Bob got this error in his project X today:</p>\n\n<p><span class=\"image-wrap\" style=\"\"><a id=\"402317_thumb\" href=\"/jira02/secure/attachment/402317/402317_image-2024-04-25-14-56-13-943.png\" title=\"image-2024-04-25-14-56-13-943.png\" file-preview-type=\"image\" file-preview-id=\"402317\" file-preview-title=\"image-2024-04-25-14-56-13-943.png\"><img src=\"https://mycompany.com/jira02/secure/thumbnail/402317/_thumb_402317.png\" style=\"border: 0px solid black\" role=\"presentation\"/></a></span></p>
```
...and this image renders perfectly inside the Description of the Improvement created in ADO.

But inside a Myprocess.Bug's _System.Description_ field all you see is:
```
<p>I tried running the following lines on a jupyter notebook (IDE= VS code),</p>\n\n<p><img src=\"/jira02/secure/attachment/402834/402834_image-2024-05-20-15-01-45-809.png\" style=\"border: 0px solid black\" /></p>\n\n<p>but I've got the following error, although the graph is still plotted after showing the error.<br/>\n<img src=\"/jira02/secure/attachment/402833/402833_image-2024-05-20-15-03-05-171.png\" style=\"border: 0px solid black\" /></p>\n\n<p>
```
Note the images are still imported and are visible in the Links section of the ADO Bugs; they just aren't rendered in the Description.

**Tool version**
3.0.442

**Comment by Alexander-Hjelm** [comment]
The first thing I notice is that the Type map seems to include the Process name. Is this correct? If the Work Item Types are named simply Bug and Improvement, then the type map should look like the following:

```json
  "type-map": {
    "type": [
      {
        "source": "Improvement",
        "target": "Improvement"
      },
      {
        "source": "Bug",
        "target": "Bug"
      }
    ]
  },
```

**Comment by cacti77** [comment]
Thanks @Alexander-Hjelm. Your suggestion was actually what I tried originally. However, even though both the jira-export and wi-import tools ran without error, no work items appeared in my DevOps project! I nearly raised an issue to ask how does one map to custom WITs; I couldn't find anything in the FAQ about this.

But when I went to https://dev.azure.com/MyOrg/MyProject/_apis/wit/workitemtypes?api-version=7.1 to list the WITs in my project I could see my custom WITs were prefixed with the process name; e.g.:
```
   {
      "name": "Bug",
      "referenceName": "Myprocess.Bug",
```
and when I updated my type-map as above the imported WITs then did appear in my ADO project, as required.

Aside: Note that the Basic process doesn't actually _have_ a Bug WIT (see [here](https://learn.microsoft.com/en-us/azure/devops/boards/work-items/guidance/choose-process?view=azure-devops&tabs=agile-process#main-distinctions-among-the-default-processes)), so maybe Samples\config-basic.json needs updating not to reference it?

Everything appears to be imported into ADO correctly for both Bugs and Improvements. The only issue is that any inline images are not rendering in the Bugs for some unexplained reason!

**Comment by Alexander-Hjelm** [comment]
Ok! Can you confirm whether or not this issue is related to https://github.com/solidify/jira-azuredevops-migrator/issues/851?

**Comment by cacti77** [comment]
Just looking at the title of that issue, I would say no, because the attached images that should appear inline in the Description are still being imported as attachments. Note too this only affects the _Description_ field of the custom Bug WIT. Any inline images in comments render perfectly well using the default mapping in the config json:
```
      {
        "source": "comment",
        "target": "System.History",
        "mapper": "MapRendered"
      },
```
I'll have a deeper look at #851 though...

**Comment by cacti77** [comment]
No, I'm pretty sure this has nothing to do with #851. Also, in my original ADO process, which also inherited from Basic, I only had one WIT type - Issue, with a custom field for Bug or Improvement (because at that time I didn't know you could add your own WITs!). The inline images in the Description field rendered fine in the ADO issues for both the 'Bug' and 'Improvement' Issues. It's only since I've created the custom WITs that this isn't happening for the new Bug WIT.

**Comment by Alexander-Hjelm** [comment]
Agreed. I addressed your point about the Bug type in the Basic template in this commit: https://github.com/solidify/jira-azuredevops-migrator/commit/3361bb8f1ef777cc7c772111878ed977ddf50d2d

Here is the CorrectDescription method for reference:

https://github.com/solidify/jira-azuredevops-migrator/blob/3361bb8f1ef777cc7c772111878ed977ddf50d2d/src/WorkItemMigrator/WorkItemImport/WitClient/WitClientUtils.cs#L442-L480

I can see that the description field seems to be hard coded for various Work Item Types. If the WI Type name is "Bug", then it seems to switch over to using ReproSteps instead of Description. Could this be the underlying issue, do you think?

**Comment by cacti77** [comment]
@Alexander-Hjelm Thank you so much for that info! Yes, my original config json above only had the one entry for the `"source": "description"` field. When I changed it to this:
```
      {
        "source": "description",
        "target": "System.Description",
	"not-for": "Myprocess.Bug",
        "mapper": "MapRendered"
      },
      {
        "source": "description",
        "target": "Microsoft.VSTS.TCM.ReproSteps",
	"for": "Myprocess.Bug",
        "mapper": "MapRendered"
      },
```
then the _Description_ field for the new Bug WIT were not populated at all once the issues were imported into ADO.
So then I changed the custom Bug WIT in ADO to add a _Repro Steps_ field and hide the _Description_ field:
![image](https://github.com/user-attachments/assets/4e6bad54-74f6-4292-9ce9-42d3ccd89d00)

Finally, when I ran the importer again the inline images rendered for both my custom Improvements (in the _Description_ field) and Bugs (in the _Repro Steps_ field). Yay, at last! (All tested using 3.0.442 still.)

I guess users just have to be careful that if they create a custom WIT called 'Bug' they have to give it a _Repro Steps_ field rather than a _Description_ field, but the former makes more sense anyway.

Thanks, too, for updating docs/Samples/config-basic.json.

Thank you once again for all your help. This tool is great, buttressed by your timely support, and has really offset the sudden pain of being kicked off a Jira server that's about to be shut down.
